### DISCUSSION TITLE
Support option for only LTS releases in datasource "endoflife-date"

### DISCUSSION THREAD

**Discussion by mikaello** [renovatebot/renovate#25363] [Suggest an Idea]
## Problem

Some packages has long term support (LTS) releases, but the information that a given package is LTS is often only available as some information in a table on the website of the package (see [prometheus website for example](https://prometheus.io/docs/introduction/release-cycle/)), while the datasource for releases have no such information (e.g. [GitHub tags for prometheus](https://github.com/prometheus/prometheus/tags)). This is often mitigated by having a manually maintained Renovate config like this:

```json
  "packageRules": [
     {
       "matchPackageNames": ["prometheus/prometheus"],
       "allowedVersions": "<2.46.0"
     }
  ]
```

## Question / possible solution

The datasource [endoflife-date](https://docs.renovatebot.com/modules/datasource/endoflife-date/) is based on  https://endoflife.date/, in [the endoflife API](https://endoflife.date/docs/api) they have the field `lts` which contains information on if this is an LTS release or not (see [prometheus releases as example](https://[endoflife.date/api/prometheus.json](https://endoflife.date/api/prometheus.json))). Would it be possible to specify an option to the `endoflife-date` datasource that it should only consider LTS-versions?

**Comment by mikaello** [comment]
An option/workaround is to create a [custom datasource](https://docs.renovatebot.com/modules/datasource/custom/) like:

```json
"customDatasources": {
  "prometheus-lts": {
    "defaultRegistryUrlTemplate": "https://endoflife.date/api/prometheus.json",
    "transformTemplates": [
      "{\"releases\": $[lts=true].{\"version\": \"v\" & latest,\"sourceUrl\": \"https://github.com/prometheus/prometheus\"},\"homepage\": \"https://prometheus.io\"}"
    ]
  }
}
```

**Comment by rarkins** [comment]
Related: https://github.com/renovatebot/renovate/issues/27632

With the above feature, it would mean "if currently on an LTS release then don't propose updates to non-LTS".

The next step would be to add LTS status to supported datasources/packages, similar in concept to the stable/unstable concept we already have.

Using your example, Renovate is currently using the github-tags datasource for prometheus/prometheus: https://github.com/prometheus/prometheus/tags

And the corresponding EOL data is https://endoflife.date/api/prometheus.json

Is the EOL website accurate and up-to-date enough to use it completely _instead_ of the github-tags? Or we'd need to blend these two? I suspect you're asking to switch to EOL in cases like this, and even if it's not perfectly up-to-date then you'd prefer to wait until they have data about a release anyway?

**Comment by thompson-tomo** [comment]
I like this however i think we should blend the data together.

The scenario i would like is:

- isLTS added to each update with default of unset and populated by endoflife.date
- isEOL added to each update with default of unset and populated by endoflife.date
- isSupported added to each update with default of unset and populated by endoflife.date
- a new updateType added eol with an update created only when the current version isEOL and the update takes it the latest version of the oldest bucket which is not eol
- The release date from endoflife.date is added to the min version in that bucket
- The latest release date from endoflife.date is added to the appropriate version
- The isEOL, isLTS & isSupported can be used in package rules

This blending is important so that the system can handle digest pinning etc and for me the new update type would be the most beneficial after setting the release dates.
